go/chat/utils.go

```diff
@@ -65,3 +66,24 @@ func Collar(lower int, ideal int, upper int) int {
 	}
 	return ideal
 }
+
+func FilterByType(msgs []chat1.MessageFromServerOrError, query *chat1.GetThreadQuery) (res []chat1.MessageFromServerOrError) {
+	if query != nil && len(query.MessageTypes) > 0 {
+		typmap := make(map[chat1.MessageType]bool)
+		for _, mt := range query.MessageTypes {
+			typmap[mt] = true
+		}
+		for _, msg := range msgs {
+			if msg.Message != nil {
+				if _, ok := typmap[msg.Message.ServerHeader.MessageType]; ok {
+					res = append(res, msg)
+				}
+			} else {
+				res = append(res, msg)
```

Comment:
I think this would make it possible to include unwanted message types if there is an unboxing error?